### PR TITLE
Only invoke completed method on unknown arguments if the failure remedy is FAIL

### PR DESCRIPTION
**Pull request by jamesnetherton**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Review by ppalaga — APPROVED** [review]

**Comment by jamesnetherton** [comment]
Converted to draft as there's something I need to check before merging.

**Comment by jamesnetherton** [comment]
@ppalaga Do you mind taking a second look at this. I made some tweaks:

- I realised that if `on-failure` is `warn`, then you need to process all valid arguments, instead of stopping on the first invalid argument, so I fixed that.

- To make the warning message output a bit friendlier, I thought it'd be best to truncate long arguments (E.g like a classpath) and abbreviate with '...'.

**Review by ppalaga — APPROVED** [review]

**Review by aldettinger — APPROVED** [review]

File: integration-tests/main-command-mode/pom.xml [hunk 1]
```diff
@@ -45,6 +45,10 @@
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-timer</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-vertx-web</artifactId>
```
Comment (aldettinger) [inline integration-tests/main-command-mode/pom.xml:50]: I guess the vertx web client is needed by the test harness.

Comment (jamesnetherton) [reply integration-tests/main-command-mode/pom.xml:50]: Correct.

**Comment by jamesnetherton** [comment]
The test failure is not related - hence merging this.

**Pull request merged**